geopandas/geodataframe.py

```diff
@@ -898,15 +898,20 @@ def iterfeatures(self, na="null", show_bbox=False, drop_id=False):
         if len(properties_cols) > 0:
             # convert to object to get python scalars.
             properties = self[properties_cols].astype(object).values
+            na_mask = pd.isna(self[properties_cols]).values
```

Comment:
```suggestion
            properties_cols = self[properties_cols]
            properties = properties_cols.astype(object).values
            na_mask = pd.isna(properties_cols).values
```

(avoid to select columns twice. At the moment, pandas will copy the data each time)